src/warzone/src/warzone/state/Startup.java

```diff
@@ -21,8 +31,174 @@ public void next() {
 		d_gameEngine.setPhase(new Reinforcement(d_gameEngine));
 	}
 	
-	 public void loadMap(String p_fileName){
-		 //todo;
+	/**
+	 * Performs the action for user command: loadmap filename
+	 * 
+	 * Game starts by user selection of a user-saved map file, which loads the map as a connected directed graph
+	 * 
+	 * @param p_fileName file name of map
+	 */
+	public void loadMap(String p_fileName){
+		 
+		String l_mapDirectory = WarzoneProperties.getWarzoneProperties().getGameMapDirectory();
+		
+		try {
+			
+			//Clear gameContext
+			d_gameContext.reset();
+
+			File l_mapFile = new File(l_mapDirectory + p_fileName);
+			
+			d_gameContext.setMapFileName(p_fileName);
+
+			//Specified file name does not exist (new map)
+			if(!l_mapFile.exists() || l_mapFile.isDirectory()) {
+				GenericView.printError("The following map file is invalid, please select another one: " + p_fileName);
+				return;
+			}
+			
+			Scanner l_scanner = new Scanner(l_mapFile);
+			String l_line;
+			String[] l_splitArray;
+			int l_continentCtr = 1;
+			int l_id;
+			Country l_country;
+
+			LoadMapPhase l_loadMapPhase = null;
+			
+			while (l_scanner.hasNextLine()) {
+				l_line = l_scanner.nextLine();
+
+				// determine which part it is
+				// file part
+				if(l_line.equals("[files]")) {
+
+					l_loadMapPhase = LoadMapPhase.FILES;
+					l_line = l_scanner.nextLine();
+				}
+				// continents part
+				else if(l_line.equals("[continents]")) {
+					
+					l_loadMapPhase = LoadMapPhase.CONTINENTS;
+					l_line = l_scanner.nextLine();
+				}
+				//countries part
+				else if (l_line.equals("[countries]")) {
+					
+					l_loadMapPhase = LoadMapPhase.COUNTRIES;
+					l_line = l_scanner.nextLine();
+				}
+				//borders part
+				else if (l_line.equals("[borders]")) {
+					
+					l_loadMapPhase = LoadMapPhase.BORDERS;
+					
+					if(!l_scanner.hasNextLine())
+						l_loadMapPhase = LoadMapPhase.COMPLETE;
+					else{
+						l_line = l_scanner.nextLine();
+					}
+				}
```

Comment:
this part can be extracted to a function : 
private LoadMapPhase  checkLoadMapPhase(){}
return updated phase if needed.